### DISCUSSION TITLE
How to spawn objects in Neighborhood?

### DISCUSSION THREAD

**Discussion by sjmarotta** [microsoft/AirSim#3930] [Support Q&A]
I'd like to be able to spawn objects via RPC at runtime during a run of Neighborhood 1.5 using the Python API. Does anyone have an example of how to to this? I am calling `simSpawnObject`, but I have no way to know what to put for `asset_name`. Looking through the code, it seems to search the `asset_map` for the name provided, and that seems to link to the Asset Registry file. But that's a binary file, and while there's some text, I don't know what, if any, of the text strings in that file correspond to asset names.

In the Python client, I'm making the following call:
```
spawn_n_val = float(random.randint(-113, 118))
spawn_e_val = 0.0
b = self.__client.simSpawnObject("new_" + str(i), "/Game/ModularNeighborhoodPack/Assets/Outdoor_Props/Car_01.Car_01", airsim.Pose(position_val=airsim.Vector3r(spawn_n_val, spawn_e_val, pos.z_val)), airsim.Vector3r(1.0, 1.0, 1.0), True)
```
(where `pos.z_val` is a known value of type `float`)

And I'm getting the following error message:
```
Traceback (most recent call last):
  File "test_add_cars.py", line 47, in <module>
    driver.start()
  File "test_add_cars.py", line 38, in start
    self.__client.simSpawnObject("new_" + str(i), "/Game/ModularNeighborhoodPack/Assets/Outdoor_Props/Car_01.Car_01",
  File "C:\Users\smarotta\Miniconda3\envs\alpaca\lib\site-packages\airsim-1.5.0-py3.8.egg\airsim\client.py", line 421, in simSpawnObject
  File "C:\Users\smarotta\Miniconda3\envs\alpaca\lib\site-packages\msgpack_rpc_python-0.4.1-py3.8.egg\msgpackrpc\session.py", line 41, in call
  File "C:\Users\smarotta\Miniconda3\envs\alpaca\lib\site-packages\msgpack_rpc_python-0.4.1-py3.8.egg\msgpackrpc\future.py", line 45, in get
msgpackrpc.error.RPCError: rpclib: function 'simSpawnObject' (called with 5 arg(s)) threw an exception. The exception is not derived from std::exception. No further information available.
```

I can only guess as to what that means. I know that I'm not giving it an asset name that it recognizes, so that's my best guess. Isi there a better way for me to peek inside the Neighborhood asset database so I can know what to feed to that parameter?

EDIT: I got it to work, eventually, through sheer luck. The `AssetRegistry.bin` file had a section toward the end that seemed to contain objects. You can see that I tried putting the long name, `/Game/ModularNeighborhoodPack/Assets/Outdoor_Props/Car_01.Car_01`, but I was still having errors. I replaced that with simply `Car_01` and it works like a champ.

HOWEVER!! I don't love having to open up the database file and hunt for strings like a barbarian. If there's a better way to know what my object options are, I'd love to know about it.

**Comment by rajat2004** [comment]
I've opened #3940 which adds an API to list all the assets. Would be great if you could try it out and see if it helps!
